src/components/organisms/teacher/OakUnitListOptionalityItem/OakUnitListOptionalityItem.tsx

```diff
@@ -0,0 +1,244 @@
+import React, {
+  ComponentPropsWithoutRef,
+  ElementType,
+  MutableRefObject,
+} from "react";
+
+import { OakUnitListOptionalityItemCard } from "../OakUnitListOptionalityItemCard";
+
+import {
+  OakFlex,
+  OakBox,
+  OakHeading,
+  OakSpan,
+  OakLabel,
+  OakGrid,
+  OakGridArea,
+} from "@/components/atoms";
+import { FlexStyleProps } from "@/styles/utils/flexStyle";
+
+const StyledYearAndOptionCount = ({
+  yearTitle,
+  optionalityUnitsLength,
+  disabledOrUnavailable,
+}: {
+  yearTitle: string | null | undefined;
+  optionalityUnitsLength: number;
+  disabledOrUnavailable: boolean | undefined;
+}) => (
+  <>
+    <OakSpan
+      $color={disabledOrUnavailable ? "text-disabled" : "text-primary"}
+      $mr={[null, "space-between-xxxl"]}
+    >
+      {yearTitle}
+    </OakSpan>
+
+    <OakBox
+      $ph={"inner-padding-xs"}
+      $pv={"inner-padding-ssx"}
+      $ba={"border-solid-s"}
+      $borderColor={
+        disabledOrUnavailable ? "border-neutral-lighter" : "bg-decorative4-main"
+      }
+      $color={disabledOrUnavailable ? "text-disabled" : "text-primary"}
+      $background={
+        disabledOrUnavailable
+          ? "bg-neutral-stronger"
+          : "bg-decorative4-very-subdued"
+      }
+      $borderRadius={"border-radius-s"}
+      $ml={[null, "space-between-s"]}
+    >
+      <OakLabel
+        $font={"heading-light-7"}
+        $color={disabledOrUnavailable ? "text-disabled" : "text-primary"}
+      >{`${optionalityUnitsLength} unit options`}</OakLabel>
+    </OakBox>
+  </>
+);
+
+const StyledIndex = ({
+  index,
+  disabledOrUnavailable,
+  ...rest
+}: {
+  index: number;
+  disabledOrUnavailable: boolean | undefined;
+} & FlexStyleProps) => (
+  <OakFlex
+    $btlr={"border-radius-s"}
+    $bblr={"border-radius-s"}
+    $background={disabledOrUnavailable ? "bg-neutral-stronger" : "lavender"}
+    $minWidth={"all-spacing-11"}
+    $width={["all-spacing-11", "all-spacing-11", "auto"]}
+    $height={["all-spacing-11", "auto", "auto"]}
+    $justifyContent={"center"}
+    $alignItems={"center"}
+    {...rest}
+  >
+    <OakSpan
+      $font={"heading-5"}
+      $color={disabledOrUnavailable ? "text-disabled" : "text-primary"}
+    >
+      {index}
+    </OakSpan>
+  </OakFlex>
+);
+
+const StyledUnitHeading = ({
+  nullTitle,
+  disabledOrUnavailable,
+}: {
+  nullTitle: string | undefined;
+  disabledOrUnavailable: boolean | undefined;
+}) => {
+  if (nullTitle === undefined) {
+    return null;
+  }
+  return (
+    <OakFlex $maxWidth={"all-spacing-21"}>
+      <OakHeading
+        $color={disabledOrUnavailable ? "text-disabled" : "text-primary"}
+        $font={"heading-7"}
+        tag="h3"
+      >
+        {nullTitle}
+      </OakHeading>
+    </OakFlex>
+  );
+};
+
+export type OakUnitListOptionalityItemProps<C extends ElementType> = {
+  as?: C;
+  disabled?: boolean;
+  unavailable?: boolean;
```

Comment:
do `disabled` and `unavailable` need to be two separate props? seems like they're always used in conjunction unless I'm missing it

Reply:
They are always used in conjunction with each other, they both come from the extended `ElementType` type props rather than component , UnitListItem uses the same pattern so i kept it consistent.

Reply:
Maybe we don't need unavailable or either ?

Reply:
Oh yeah I see, yeah it seems like we could just condense down to 1 or the other? save some confusion!

Reply:
I guess unavailable is if it is expired and disabled could be copyright / no logged in. But the style is the same atm so i say we stick with unavailable ?